.github/workflows/pytest_postgres.yml

```diff
@@ -0,0 +1,87 @@
+name: Run postgres tests with cache
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version: [ "3.7.15", "3.8.16", "3.9.10" ]
```

Comment:
Do we need to test it against all three versions?

Reply:
Not necessarily - just had this copied from the main set of tests and wanted to err on the side of testing more rather than less.

Reply:
This was also more as an initial demonstration that things broadly work - think it might be worth rejigging how the CI tests are set up to split the different backends up a bit possibly